### PR TITLE
Add json prettyprinting detail view to the log details page.

### PR DESCRIPTION
**Pull request by DaveShuckerow**
Also adds in a crossfade animation between log details views.

image

I'm going to punt on the error inspector for a few weeks to prepare for demos.

### CODE REVIEW

File: packages/devtools_app/lib/src/logging/flutter/logging_screen.dart [hunk 1]
```diff
@@ -98,6 +113,113 @@ class LogsTable extends StatelessWidget {
...
+class _LogDetailsState extends State<LogDetails>
+    with SingleTickerProviderStateMixin {
+  AnimationController crossFade;
+
+  @override
+  void initState() {
+    super.initState();
+    crossFade = AnimationController(
+      vsync: this,
+      duration: const Duration(milliseconds: 200),
```
Comment (jacob314) [inline packages/devtools_app/lib/src/logging/flutter/logging_screen.dart]: we should soon do an audit of all durations used and add some top level duration constants used throughout the app.

File: packages/devtools_app/lib/src/logging/logging_controller.dart [hunk 2]
```diff
@@ -656,17 +658,28 @@ class LogData {
 
   final RemoteDiagnosticsNode node;
   String _details;
-  Future<String> detailsComputer;
+  Future<String> Function() detailsComputer;
+
+  static const JsonEncoder prettyPrinter = JsonEncoder.withIndent('    ');
```
Comment (jacob314) [inline packages/devtools_app/lib/src/logging/logging_controller.dart]: why indent 4? use indent 2 and a fixed sized font instead. There is some similar code in the json display in the timeline.

File: packages/devtools_app/lib/src/logging/flutter/logging_screen.dart [hunk 3]
```diff
@@ -68,13 +72,23 @@ class _LoggingScreenState extends State<LoggingScreenBody> {
         ],
       ),
       Expanded(
-        child: LogsTable(
-          data: controller.data,
+        child: Split(
```
Comment (jacob314) [inline packages/devtools_app/lib/src/logging/flutter/logging_screen.dart]: do we not think expandable table children are practical? showing a separate details view in the existing logging page was more for expediency than by design. I'd hope we can do better in Flutter and actually show the details inline.

File: packages/devtools_app/lib/src/logging/logging_controller.dart (hunk 2 above)
Comment (DaveShuckerow) [reply packages/devtools_app/lib/src/logging/logging_controller.dart]: We don't have any monospaced fonts bundled yet.  I've left a TODO.

File: packages/devtools_app/lib/src/logging/flutter/logging_screen.dart (hunk 3 above)
Comment (DaveShuckerow) [reply packages/devtools_app/lib/src/logging/flutter/logging_screen.dart]: It is doable, just out of scope for this PR, as discussed offline.

**Comment by jacob314** [comment]
lgtm

**Pull request merged**